### PR TITLE
Implement ConsolePlugin reconciler

### PR DESCRIPTION
**Pull request by mresvanis**
This PR implements the ConsolePlugin reconciler, in order for the NVIDIA GPU Addon Operator to be able to deploy the [Console Plugin NVIDIA GPU](https://github.com/rh-ecosystem-edge/console-plugin-nvidia-gpu) when:

- the plugin boolean field is enabled in the GPUAddon CR (which is enabled by default)
- the OpenShift version is `>= 4.10.*`

### Changes

The [deployment](https://github.com/rh-ecosystem-edge/console-plugin-nvidia-gpu/tree/main/deployment/console-plugin-nvidia-gpu/templates) of the Console Plugin NVIDIA GPU includes:

- a Deployment
- a Service
- a [ConsolePlugin](https://github.com/openshift/api/blob/6f735e7109c87826edee3ca02a753071ecc933b9/console/v1alpha1/types_console_plugin.go#L14) CR
- patches the `cluster` console in order to enable the console plugin

In addition:

- the operator cluster-role is extended to include `ConsolePlugin` resources (since it's a cluster-scoped resource)
- the operator cluster-role is extended to include `Console` resources (in order to enable the plugin)
- the operator role is extended to include `Deployment` and `Service` resources
- the controller is setup to watch its own `ConsolePlugins`, `Deployments` and `Services`
- test coverage got also a big higher:

```shell
# From
ok      github.com/rh-ecosystem-edge/nvidia-gpu-addon-operator/controllers      0.034s  coverage: 68.6% of statements
ok      github.com/rh-ecosystem-edge/nvidia-gpu-addon-operator/internal/common  0.029s  coverage: 65.0% of statements
# To
ok      github.com/rh-ecosystem-edge/nvidia-gpu-addon-operator/controllers      0.034s  coverage: 71.4% of statements
ok      github.com/rh-ecosystem-edge/nvidia-gpu-addon-operator/internal/common  0.023s  coverage: 67.3% of statements
```

### CODE REVIEW

File: controllers/consoleplugin_resource_reconciler.go
Comment (fabiendupont) [inline controllers/consoleplugin_resource_reconciler.go:43]: Any reason why it's not mentioning nvidia?

Comment (mresvanis) [reply controllers/consoleplugin_resource_reconciler.go:43]: No reason at all, I'll update to `ocp-nvidia-gpu-addon` and also open an additional PR to change the same name in the default config values: https://github.com/rh-ecosystem-edge/nvidia-gpu-addon-operator/blob/main/internal/common/common.go#L24

**Comment by mresvanis** [comment]
There is a few manual tests that I would like to perform before merging this PR. 

/hold

**Review by fabiendupont — APPROVED** [review]

**Pull request merged**
